Guide.docc/CommonProblems.md

```diff
@@ -719,3 +719,65 @@ class WindowStyler {
 All `Sendable` properties can still be safely accessed in this `init` method.
 And while any non-`Sendable` properties cannot,
 they can still be initialized by using default expressions.
+
+### Non-Isolated Deinitialization
+
+By default, any custom initializers you write will have the same isolation as
+their enclosing scope.
+This is _not_ the case for deinitializers, which are always non-isolated.
+
+```swift
+@MainActor
+class WindowStyler {
+    // A MainActor-isolated type
+    private let store = StyleStore()
+
+    init {
+        // infers MainActor
+    }
+
+    deinit {
+        store.stopNotifications()
+    }
+}
+```
+
+This code produces the error:
+
+```
+error: call to main actor-isolated instance method 'stopNotifications()' in a synchronous nonisolated context
+ 5 |     
+ 6 |     deinit {
+ 7 |         store.stopNotifications()
+   |               `- error: call to main actor-isolated instance method 'stopNotifications()' in a synchronous nonisolated context
+ 8 |     }
+ 9 | }
+```
+
+While this might feel surprising, given that this type is `MainActor`-isolated,
+this is not a new constraint.
+The thread that executes a deinitializer has never been guaranteed and
+Swift's data isolation is now just surfacing that fact.
+
+Often, the work being done within the `deinit` does not need to be synchronous.
+A solution is to use an unstructured `Task` to first capture and
+then operate on the isolated values.
+When using this technique,
+it is _critical_ to ensure you do not capture `self`, even implicitly.
+
+```swift
+@MainActor
+class WindowStyler {
+    // A MainActor-isolated type
+    private let store = StyleStore()
+
+    deinit {
+        Task { [store] in
```

Comment:
Maybe worth pointing out that Task is NOT main actor isolated
```suggestion
        Task { [store] in // not actor isolated, because since definit is not either
```

Reply:
or maybe make a Task.detached here even tbh?

Reply:
I bet making this clearer with a comment will be easier for most readers to digest than using `Task.detached`, even if that is more explicit semantically.